dockerfiles/theia/ubi8-nodejs10.Dockerfile

```diff
@@ -0,0 +1,152 @@
+# Copyright (c) 2018 Red Hat, Inc.
+# This program and the accompanying materials are made
+# available under the terms of the Eclipse Public License 2.0
+# which is available at https://www.eclipse.org/legal/epl-2.0/
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+# Contributors:
+#   Red Hat, Inc. - initial API and implementation
+
+###
+# Builder Image
+#
+# Should be based on CentOS
```

Comment:
I see no reason to have it based on centos

Reply:
Thanks for pointing out. The comment is not clear enough. I meant, that if we build Che Theia on UBI we have to use CentOS based Che-Theia-dev

Reply:
Otherwise Theia might not work

Reply:
I still don't see why

we're able to use debian in e2e images (which is not alpine based)
https://github.com/eclipse/che-theia/blob/master/dockerfiles/theia/e2e/Dockerfile#L10

Reply:
we're doing typescript so it should be portable across all systems

Reply:
No, unfortunately it isn't. We have some native code in node side dependencies. That's the reason why I've chosen CentOS but not Debian. Generally speaking, it should have the same base distro for builder and runner container (well, it may work, but no guarantee on that).

Reply:
@mmorhun what are these dependencies ? it's not the node-gyp dependencies ?
AFAIK we're rebuilding them there https://github.com/eclipse/che-theia/blob/master/dockerfiles/theia/e2e/Dockerfile#L31 for example

Reply:
Thanks, we need to try